src/transformers/models/rt_detr/modeling_rt_detr.py

```diff
@@ -1736,7 +1736,7 @@ def generate_anchors(self, spatial_shapes=None, grid_size=0.05, device="cpu", dt
         anchors = torch.concat(anchors, 1)
         valid_mask = ((anchors > eps) * (anchors < 1 - eps)).all(-1, keepdim=True)
         anchors = torch.log(anchors / (1 - anchors))
-        anchors = torch.where(valid_mask, anchors, torch.inf)
+        anchors = torch.where(valid_mask, anchors, torch.tensor(float("inf"), dtype=torch.float32, device=device))
```

Comment:
```suggestion
        anchors = torch.where(valid_mask, anchors, torch.tensor(float("inf"), dtype=dtype, device=device))
```

Reply:
2ea9b757c2217344da37173fde41eb9a1be66496

Reply:
Added some doc f15740647f341555e65720e942586419ad481d45 :) Tell me what do you think